src/pip/_internal/req/req_install.py

```diff
@@ -633,6 +634,18 @@ def update_editable(self, obtain=True):
         vc_type, url = self.link.url.split('+', 1)
         vcs_backend = vcs.get_backend(vc_type)
         if vcs_backend:
+            if not self.link.is_vcs:
+                reason = (
+                    "This form of VCS URL is being deprecated: {}."
+                ).format(
+                    self.link.url
+                )
+                replacement = None
+                if self.link.url.startswith("git+git@"):
+                    replacement = (
+                        "git+git:// or git+ssh://"
+                    )
```

Comment:
I would suggest `git+https://` instead. `git+git://` should also be discouraged (see #1983), and all popular, recent Git server should support the “smart” protocol over HTTPS, which is guaranteed to be secure.

Reply:
@uranusjr good point, done.